javascript/portfolio.js

```diff
@@ -1,5 +1,6 @@
 'use strict';
 var portObjectArray = [];
+var ports = {};
```

Comment:
You are not using this. I would suggest you do, similar to lab in our view js file and put your call to .toHtml in its own method as part of this object. Or delete it.